### PR TITLE
Remove support for legacy menu markup

### PR DESCRIPTION
**Pull request by CelticMinstrel**
Since it was removed from `[message][option]` already, perhaps we should remove it entirely. After all, the deprecation message does say it'll be removed in 1.15.x.

### CODE REVIEW

**Comment by Wedge009** [comment]
Purge away!

File: src/gui/dialogs/campaign_difficulty.cpp [hunk 1]
```diff
@@ -77,35 +76,6 @@ config generate_difficulty_config(const config& source)
 	// Populate local config with difficulty children
 	result.append_children(source, "difficulty");
 
-	// Convert legacy format to new-style config if latter not present
-	if(result.empty() && source.has_attribute("difficulties")) {
-		deprecated_message("[campaign]difficulties", DEP_LEVEL::FOR_REMOVAL, {1, 15, 0}, "Use [difficulty] instead.");
```
Comment (jostephd) [inline src/gui/dialogs/campaign_difficulty.cpp]: Surely the deprecated_message function call should be retained, and upgraded to DEP_LEVEL::REMOVED?

**Comment by CelticMinstrel** [comment]
I mean, sure, it _could_ be, but the intent of DEP_LEVEL::REMOVED is for things that were removed _without_ deprecation. My understanding of the deprecation process was that once deprecated functionality is removed, the warning doesn't need to be kept.

**Comment by jostephd** [comment]
My understanding of https://wiki.wesnoth.org/CompatibilityStandards#Post-removal is that a message at DEP_LEVEL::REMOVED should be retained for one stable version (so, kept in 1.15.x and 1.16.x but removed in 1.17.x). Keeping the message would make it easier to port add-ons from 1.12 to 1.16 without having to first install 1.14. (This is particularly helpful for users of LTS distros, who sometimes skip an entire stable version because of how wesnoth's and distros' release schedules align.)

**Comment by CelticMinstrel** [comment]
Ah, fair enough... I have a feeling we've violated that in a number of other places too. Feel free to push a new error message to this branch if you have time (taking care to cover all locations); if not, I'll see when I can get it done myself.

**Comment by jostephd** [comment]
There were three instances. I restored two in 18a349fa2d4e126279d3ac4629fbd36ace61bf0d, but wasn't sure what best to do about the one in the legacy_menu_item constructor.

**Comment by CelticMinstrel** [comment]
Thanks! I'll merge this, and if you think of a better way to handle the other case, you can always add that in later.

**Pull request merged**